python/pyspark/sql/dataframe.py

```diff
@@ -347,13 +353,18 @@ def show(self, n=20, truncate=True, vertical=False):
          name | Bob
         """
         if isinstance(truncate, bool) and truncate:
-            print(self._jdf.showString(n, 20, vertical))
+            print(self._jdf.showString(n, 20, vertical, False))
         else:
-            print(self._jdf.showString(n, int(truncate), vertical))
+            print(self._jdf.showString(n, int(truncate), vertical, False))
 
     def __repr__(self):
         return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
 
+    def _repr_html_(self):
+        if self._eager_eval:
+            return self._jdf.showString(
+                self._default_console_row, self._default_console_truncate, False, True)
```

Comment:
We need to return `None` if `self._eager_eval` is `False`.

Reply:
```
What will be shown if spark.jupyter.eagerEval.enabled is False? Fallback the original automatically?
```
Yes, it will fallback to call __repr__.

```
We need to return None if self._eager_eval is False.
```
Got it, more clear in code logic.